src/Nethermind/Nethermind.Monitoring/Metrics/MetricsController.cs

```diff
@@ -52,16 +54,30 @@ public void RegisterMetrics(Type type)
 
         private static Gauge CreateMemberInfoMetricsGauge(MemberInfo member)
         {
-            Dictionary<string, string> staticLabels = member
-                .GetCustomAttributes<MetricsStaticDescriptionTagAttribute>()
-                .ToDictionary(
-                    attribute => attribute.Label,
-                    attribute => GetStaticMemberInfo(attribute.Informer, attribute.Label));
+            Dictionary<string, string> staticTags = GetCommonStaticTags();
+
+            member.GetCustomAttributes<MetricsStaticDescriptionTagAttribute>()
+                  .ForEach(attribute => staticTags.Add(attribute.Label, GetStaticMemberInfo(attribute.Informer, attribute.Label)));
 
             string description = member.GetCustomAttribute<DescriptionAttribute>()?.Description;
             string name = BuildGaugeName(member);
 
-            return CreateGauge(name, description, staticLabels);
+            return CreateGauge(name, description, staticTags);
+        }
+
+        // Tags that all metrics share
+        private static Dictionary<string, string> GetCommonStaticTags()
```

Comment:
You are creatign this dictionary for every Gauge. If there are no `MetricsStaticDescriptionTagAttribute` (almost all?) let's reuse one dictionary